app/controllers/customers_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CustomersController < ApplicationController
+
+  def new
+  end
+
+  def create
+    customer = Customer.new(customer_params)
+    if customer.save
+      session[:customer_id] = customer.id
+      redirect_to 'root_path'
```

Comment:
this should not be a string! it is causing your tests to not pass and is breaking the app.